File: examples/python/core/steps/test-retry-behavior.py
@pytest.mark.durable_execution(
    handler=handler_with_retry,
    lambda_function_name="retry_function",
)
def test_step_retry(durable_runner):
    """Test step retry behavior."""
    with durable_runner:
        result = durable_runner.run(input={}, timeout=30)
    
    # Function should eventually succeed after retries
    assert result.status is InvocationStatus.SUCCEEDED
    
    # Inspect the step that retried
    step_result = result.get_step("unreliable_operation")
    assert step_result.status is InvocationStatus.SUCCEEDED

File: examples/java/core/steps/test-retry-behavior.java
@Test
void testStepRetry() {
    var runner = LocalDurableTestRunner.create(Map.class, new RetryHandler());

    var result = runner.run(Map.of());

    // Function should eventually succeed after retries
    assertEquals(ExecutionStatus.SUCCEEDED, result.getStatus());

    // Inspect the step that retried
    var stepOp = runner.getOperation("unreliable-operation");
    assertEquals(ExecutionStatus.SUCCEEDED, stepOp.getStatus());
}
